scripts/globals/items/chaosbringer.lua

```diff
@@ -9,17 +9,16 @@ itemObject.onItemDrop = function(player, item)
 end
 
 itemObject.onItemEquip = function(player, item)
-    player:addListener("ENGAGE", "CHAOSBRINGER_KILLS", function(playerArg, mob)
-        if mob:getLocalVar("listenerApplied") == 0 then
-            mob:addListener("TAKE_DAMAGE", "VALID_KILL", function(mobArg, amount, attacker, attackType, damageType)
-                mobArg:setLocalVar("listenerApplied", 1)
+    player:addListener("ATTACK", "CHAOSBRINGER_KILLS", function(playerAttkListener, mobAttkListener)
+        if mobAttkListener:getLocalVar("CBListenerApplied") == 0 then
+            mobAttkListener:addListener("ATTACKED", "VALID_KILL", function(mobAttkedListener, playerAttkedListener, action)
+                mobAttkedListener:setLocalVar("CBListenerApplied", 1)
                 if
-                    attacker:getQuestStatus(xi.quest.log_id.BASTOK, xi.quest.id.bastok.BLADE_OF_DARKNESS) >= QUEST_ACCEPTED and
-                    attacker:getEquipID(xi.slot.MAIN) == xi.items.CHAOSBRINGER and
-                    attackType == xi.attackType.PHYSICAL and
-                    amount > mobArg:getHP()
+                    playerAttkedListener:getQuestStatus(xi.quest.log_id.BASTOK, xi.quest.id.bastok.BLADE_OF_DARKNESS) >= QUEST_ACCEPTED and
+                    playerAttkedListener:getEquipID(xi.slot.MAIN) == xi.items.CHAOSBRINGER and
+                    mobAttkedListener:getHP() == 0
```

Comment:
WS kills shouldnt count correct, just auto attacks? Is it somewhere else that determines if the kill was from a WS or regular attack?

Reply:
> WS kills shouldnt count correct, just auto attacks? Is it somewhere else that determines if the kill was from a WS or regular attack?

The attack and attacked listener are only called on the OnAttack function which is just for normal physical attacks.

Reply:
Can players use ranged attacks to exploit this? That was a big issue before I mangled this code together to work the way it did.

Reply:
> Can players use ranged attacks to exploit this? That was a big issue before I mangled this code together to work the way it did.

Nope, the attack and attacked listener are only called on the OnAttack function which is just for normal physical attacks and not ranged attacks which use a different function.